src/App.js

```diff
@@ -1,21 +1,88 @@
 import "./index.css";
-import { Routes, Route, Link } from "react-router-dom";
-import { Home, CategoryList, CategoryDetails, VideoDetails } from "./pages";
-import { Navbar } from "./components";
+import { Routes, Route } from "react-router-dom";
+import {
+  Home,
+  VideoDetails,
+  Playlists,
+  PlaylistDetails,
+  Login,
+  Signup,
+  Account,
+} from "./pages";
+import { Navbar, Footer } from "./components";
+import { useEffect, useState } from "react";
+import axios from "axios";
+import { useAuth, useData } from "./context";
+import { PrivateRoute } from "./api/ProtectedRoute";
 
 function App() {
+  const { videos, dispatch } = useData();
+  const { user } = useAuth();
+  const [theme, setTheme] = useState("dark");
+
+  const fetchVideosData = async () => {
+    try {
+      const response = await axios.get(
```

Comment:
Extract `Axios` call into a function, with error management. Would help decrease complexity, and reusability will increase.